tests/phpunit/tests/admin/includesPost.php

```diff
@@ -130,6 +136,8 @@ function test__wp_translate_postdata_cap_checks_editor() {
 	 * edit_post() should convert an existing auto-draft to a draft.
 	 *
 	 * @ticket 25272
+	 *
+	 * @covers ::wp_insert_post
```

Comment:
```suggestion
	 * @covers ::edit_post
```

Technically you're right, the change happens in `wp_insert_post` as it sets the default value when not passed to it. However, `edit_post` is the function that's setting up the post data and then invokes `wp_insert_post`. In this case, `post_status` is not set in the post data, which then gets set in `wp_insert_post` as the default of `draft`. 

So this test is specifically testing what happens when invoking `edit_post` without passing the `post_status` parameter it. This makes this test an integration test.